src/libraries/Microsoft.Extensions.Options/src/UnnamedOptionsManager.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.Extensions.Options
+{
+    internal class UnnamedOptionsManager<[DynamicallyAccessedMembers(Options.DynamicallyAccessedMembers)] TOptions> :
+        IOptions<TOptions>
+        where TOptions : class
+    {
+        private readonly Lazy<TOptions> _lazy;
+
+        public UnnamedOptionsManager(IOptionsFactory<TOptions> factory)
+        {
+            _lazy = new Lazy<TOptions>(() => factory.Create(Options.DefaultName));
+        }
+
+        public TOptions Value => _lazy.Value;
+    }
```

Comment:
Looks simple enough.

Can you remind me what costs this is trying to address?

Reply:
The current implementation is shared between "named" and "unnamed" Options. When using "named" options, we need to use a cache (ConcurrentDictionary):

https://github.com/dotnet/runtime/blob/c37395208c35770a0203785b886f990ee5c12fcd/src/libraries/Microsoft.Extensions.Options/src/OptionsManager.cs#L29-L49

In the simple case where you never use a named option, we still create the dictionary, and check it to get the value.

Here, when you never use a named option, we can simply use a Lazy, saving the dictionary alloc and access.

Reply:
Is it better to use LazyInitializer and keep the state in the implementation?

Reply:
Ah I re-read your statement, I thought you were saying the ConcurrentDictionary was cheaper. You're just saying they are cheaper ways to avoid the lazy overhead. I thought about it a little but figured I'd let you comment and tell me what the optimal way was 😉

Reply:
https://github.com/dotnet/runtime/pull/49883